terraform/modules/hetzner-kubernetes/machines.variables.tf

```diff
@@ -0,0 +1,35 @@
+variable "default_location" {
+  default = "nbg1"
+}
+
+variable "default_server_type" {
+  default = "cx51"
+}
+
+variable "default_image" {
+  default = "ubuntu-18.04"
+}
+
+
+# FUTUREWORK: replace 'any' by implementing https://www.terraform.io/docs/language/functions/defaults.html
+#
+variable "machines" {
+  description = "list of machines"
+  # type = list(object({
+  #   group_name = string
+  #   machine_id = string
+  #   machine_type = string
+  #   component_classes = list(string)
```

Comment:
As I understand there are 2 valid component classes: "controlplane" and "node" and terraform provides no way of doing enums. So, can we do the next best thing which would be to document and validate this?

I have a feeling that we are over-generalizing here. Nobody wants a "machine" in their kubernetes cluster. Maybe it is better to accept "pools" for nodes and controlplane separately. This will also get rid of the enum problem.

Reply:
This seems to be the same concern as I expressed here https://github.com/wireapp/wire-server-deploy/pull/418#issuecomment-776606347

My biggest concern is that we are giving too much freedom. E.g. nothing is stopping you from accidentally assigning `2` or `4` control-plane nodes; whilst those setups do not work in etcd.  you should only have `1` or `3` control-plane nodes and _in very rare cases_ where you have many regions or zones, an etcd cluster of `5` is possible (but has bad performance. As every write needs consensus of 4 nodes instead instead of 2  )


I would still suggest us having two sections:

```
control_plane =  {
  count  = 3  # can be 3 or 1; but no other value
  is_worker = true  # This sets the component_class label on the control_plane nodes
}

node_pools = [
{ count = 200 name = "basic",  machine_type = blah }
{ count = blah,  machine_type = blah }
]
```